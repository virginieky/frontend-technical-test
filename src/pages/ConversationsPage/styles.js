import styled from 'styled-components';
import PropTypes from 'prop-types';

export const Container = styled.div`
  position: relative;
  height: 100%;
  width: 100%;
  max-width: 1024px;
  margin: 0 auto;
`;

export const BackButton = styled.button`
  position: absolute;
  top: 20px;
  left: 20px;
  z-index: 4;
  background: white;
  border: 0;
  outline: 0;
  display: ${({ isHidden }) => (!isHidden ? 'block' : 'none')};
  @media (min-width: 1024px) {
    display: none;
  }
`;

export const Panel = styled.div`
  height: 100%;
  width: 100%;
  background: #f5f5f5;
  @media (min-width: 1024px) {
    width: 50%;
    display: inline-block;
    vertical-align: top;
  }
`;

export const RightPanel = styled(Panel)`
  z-index: 0;
`;

export const LeftPanel = styled(Panel)`
  position: absolute;
  top: 0;
  left: 0;
  z-index: 9;
  transition: transform 0.3s ease-in-out;
  transform: ${({ isHidden }) =>
    !isHidden ? 'translateX(0)' : 'translateX(-100%)'};

  @media (min-width: 1024px) {
    position: relative;
    transition: initial;
    transform: initial;
    border-left: 1px solid #e9e9e9;
  }
`;

LeftPanel.defaultProps = {
  isHidden: 'false',
};

LeftPanel.propTypes = {
  isHidden: PropTypes.bool,
};
