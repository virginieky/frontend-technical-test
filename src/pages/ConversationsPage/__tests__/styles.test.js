import React from 'react';
import { render } from '@testing-library/react';
import { BackButton, LeftPanel } from '../styles';

describe('pages | ConversationsPage | components', () => {
  describe('BackButton', () => {
    it('should match the snapshot if isHidden is false', () => {
      const { asFragment } = render(<BackButton isHidden={false} />);

      expect(asFragment()).toMatchSnapshot();
    });

    it('should match the snapshot if isHidden is true', () => {
      const { asFragment } = render(<BackButton isHidden={true} />);

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('LeftPanel', () => {
    it('should match the snapshot if isHidden is false', () => {
      const { asFragment } = render(<LeftPanel isHidden={false} />);

      expect(asFragment()).toMatchSnapshot();
    });

    it('should match the snapshot if isHidden is true', () => {
      const { asFragment } = render(<LeftPanel isHidden={true} />);

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
