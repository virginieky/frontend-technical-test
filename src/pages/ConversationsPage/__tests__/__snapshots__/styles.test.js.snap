// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages | ConversationsPage | components BackButton should match the snapshot if isHidden is false 1`] = `
<DocumentFragment>
  <button
    class="styles__BackButton-sc-1444re7-1 ecAQdg"
  />
</DocumentFragment>
`;

exports[`pages | ConversationsPage | components BackButton should match the snapshot if isHidden is true 1`] = `
<DocumentFragment>
  <button
    class="styles__BackButton-sc-1444re7-1 masfD"
  />
</DocumentFragment>
`;

exports[`pages | ConversationsPage | components LeftPanel should match the snapshot if isHidden is false 1`] = `
<DocumentFragment>
  <div
    class="styles__Panel-sc-1444re7-2 styles__LeftPanel-sc-1444re7-4 lhqZAk cnzVEV"
  />
</DocumentFragment>
`;

exports[`pages | ConversationsPage | components LeftPanel should match the snapshot if isHidden is true 1`] = `
<DocumentFragment>
  <div
    class="styles__Panel-sc-1444re7-2 styles__LeftPanel-sc-1444re7-4 lhqZAk cRreCo"
  />
</DocumentFragment>
`;
